clients/lsp-rust.el

```diff
@@ -860,6 +866,45 @@ them with `crate` or the crate name they refer to."
        (lambda (_) (concat "*" label "*")))
       (setq lsp-rust-analyzer--last-runnable runnable))))
 
+(defun lsp-rust-analyzer-debug (runnable)
+  "Select and run a runnable action."
+  (interactive (list (lsp-rust-analyzer--select-runnable)))
+  (unless (featurep 'dap-cpptools)
```

Comment:
Wouldn't it be better to `require` `dap-cpptools` here instead of just throwing an error if the user hasn't done so already? As a bonus, `require` throws an error if it isn't installed itself, but this can be wrapped in an intuitive error-message.

Reply:
I am not quite sure about that... My concern that we have several adapters with :type = "lldb". ("cppdbg" is actually wrong, should be "lldb").

Reply:
`counsel-projectile-rg`ing dap-mode, I don't see `dap-register-debug-provider "lldb" anywhere other than `dap-gdb-lldb.el`. Or do you propose renaming `cppdbg` to `lldb` in the future? Isn't `cpptools` for GDB? This PR would have to be adapted in the latter case anyway, though.

Reply:
type is set to "lldb" in dap-codelldb as well.

> Or do you propose renaming cppdbg to lldb in the future?

I think we should at some point, otherwise, we won't work with launch.json coming from vscode.